src/controller/CHIPDeviceController.h

```diff
@@ -96,6 +98,21 @@ class DLL_EXPORT DevicePairingDelegate
     virtual void OnPairingDeleted(CHIP_ERROR error) {}
 };
 
+class DLL_EXPORT ScriptDevicePairingDelegate final : public DevicePairingDelegate
+{
+public:
+    ~ScriptDevicePairingDelegate() = default;
+    void SetWifiCredential(const char * ssid, const char * password);
+    void OnNetworkCredentialsRequested(RendezvousDeviceCredentialsDelegate * callback) override;
+
+    void OnOperationalCredentialsRequested(const char * csr, size_t csr_length,
+                                           RendezvousDeviceCredentialsDelegate * callback) override;
+
+private:
+    // WiFi Provisioning Data
+    std::string wifiSSID, wifiPassword;
```

Comment:
split as two lines.
Names are prefixed with 'm', like mWifiSsid mWifiPassword.

std::string potentially uses heap. I am unsure what the usage for this class is - is it allowed to use heap?

Reply:
we should avoid std::string where-ever possible, whether we think the environment will have heap or not.

these items have known maximum lengths (32 and 63) right? we don't need to support arbitrary null-terminated strings.

Reply:
we should avoid std::string where-ever possible, whether we think the environment will have heap or not.

these items have known maximum lengths (32 and 63) right? we don't need to support arbitrary null-terminated strings.